### PR TITLE
Add documentation for user categories endpoint

### LINKED ISSUE #427: Document User Categories API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by joshsmith**
Closes #427.

### CODE REVIEW

File: blueprint/api.apib [hunk 1]
```diff
@@ -386,6 +386,108 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
+
++ Request
+
+    + Attributes
+
+      + data (object)
+
+          + relationships (object)
+
+              + category (object)
```
Comment (christopherstyles) [inline blueprint/api.apib:421]: I’m finding a newline between each attribute to be a little difficult to read. I think it’s safe to group attributes together w/o newlines. Thoughts? We should also probably update the RFC data structures section to reflect 4-space indentation…

Comment (joshsmith) [inline blueprint/api.apib:421]: Yes, there's a number of things that need updated in that RFC, and it should be closed ASAP with an actual PR for docs of docs...of docs?

File: blueprint/api.apib [hunk 2]
```diff
@@ -386,6 +386,103 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
+
++ Response 401 (application/json)
+
+    + Attributes (OAuth Invalid Response)
+
++ Response 422 (application/vnd.api+json)
+
+    + Attributes (Unprocessable Entity Response)
+
+## User Category [/organizations/{id}]
```
Comment (christopherstyles) [inline blueprint/api.apib]: Should this be `/user_categories/{id}` instead of `/organizations/{id}`?

Comment (joshsmith) [inline blueprint/api.apib]: No, I'm pretty sure it should be organizations...

File: blueprint/api.apib [hunk 3]
```diff
@@ -386,6 +386,103 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
 
 # Group User Categories
 
+This resource identifies a relationship between a User and a Category. For example, the category of "Society" may belong to a given user.
+
+It may be eaasier to think of a User Category as one of a user's interest. They may be interested in "Society", "Technology" and "Government". We use those interests to match users up to relevant organizations and projects. The term "category" therefore is more appropriate for _categorizing_ those organizations and projects.
+
+## User Categories [/user_categories]
+
+### Filters user categories by id [GET]
```
Comment (christopherstyles) [inline blueprint/api.apib:395]: Should this be singular declarative? i.e. `Filter user categories…`

**Pull request closed without merging**